src/main/resources/db/migration/0300/V320__AdHocPlotObservations.sql

```diff
@@ -0,0 +1,11 @@
+CREATE TABLE tracking.observation_types (
+    id INTEGER PRIMARY KEY,
+    name TEXT NOT NULL UNIQUE
+);
+
+ALTER TABLE tracking.observations ADD COLUMN is_ad_hoc BOOLEAN DEFAULT FALSE NOT NULL;
```

Comment:
Opinions differ on this, but I prefer not using `DEFAULT` for this kind of thing because it eliminates a little bit of automated bug detection: if, somewhere in the server code, we accidentally forget to set the `is_ad_hoc` value when we're inserting a new row, the insertion will succeed and use the default, rather than failing because we didn't specify a value for a mandatory column.

That's why a lot of my migrations look like "add nullable column, populate existing rows, set it to non-nullable."

That said, `DEFAULT` is part of SQL for a reason. If we want "forgot to include the column when inserting" cases to make a best-effort attempt to insert the row rather than abort the whole transaction, this is how to do it.

Reply:
Makes sense. I was doing this as a means to save on modification of the pre-existing observations code, but I see the benefit in explicitly migrating the field.